src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -442,6 +442,1211 @@ Sum: 10, Count: 4")
 
         End Sub
 
+        <Fact()>
+        Public Sub Overloading001()
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb"><![CDATA[
+
+Module m1
+    Sub Test(x as (a as integer, b as Integer))
+    End Sub
+
+    Sub Test(x as (c as integer, d as Integer))
+    End Sub
+End module
+
+]]></file>
+</compilation>, additionalRefs:={ValueTupleRef, SystemRuntimeFacadeRef})
+
+            comp.AssertTheseDiagnostics(
+<errors>
+    BC30269: 'Public Sub Test(x As (a As Integer, b As Integer))' has multiple definitions with identical signatures.
+    Sub Test(x as (a as integer, b as Integer))
+        ~~~~
+</errors>)
+
+        End Sub
+
+        <Fact()>
+        Public Sub Overloading002()
+            Dim comp = CreateCompilationWithMscorlibAndVBRuntime(
+<compilation>
+    <file name="a.vb"><![CDATA[
+
+Module m1
+    Sub Test(x as (integer,Integer))
+    End Sub
+
+    Sub Test(x as (a as integer, b as Integer))
+    End Sub
+End module
+
+]]></file>
+</compilation>, additionalRefs:={ValueTupleRef, SystemRuntimeFacadeRef})
+
+            comp.AssertTheseDiagnostics(
+<errors>
+BC30269: 'Public Sub Test(x As (Integer, Integer))' has multiple definitions with identical signatures.
+    Sub Test(x as (integer,Integer))
+        ~~~~
+</errors>)
+
+        End Sub
+
+        <Fact()>
+        Public Sub SimpleTupleTargetTyped001()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Imports System
+Module C
+
+    Sub Main()
+        Dim x as (String, String) = (Nothing, Nothing)
+        System.Console.WriteLine(x.ToString())
+    End Sub
+End Module
+
+    </file>
+</compilation>, additionalRefs:={ValueTupleRef, SystemRuntimeFacadeRef}, expectedOutput:=<![CDATA[
+(, )
+            ]]>)
+
+            verifier.VerifyIL("C.Main", <![CDATA[
+{
+  // Code size       28 (0x1c)
+  .maxstack  3
+  .locals init (System.ValueTuple(Of String, String) V_0) //x
+  IL_0000:  ldloca.s   V_0
+  IL_0002:  ldnull
+  IL_0003:  ldnull
+  IL_0004:  call       "Sub System.ValueTuple(Of String, String)..ctor(String, String)"
+  IL_0009:  ldloca.s   V_0
+  IL_000b:  constrained. "System.ValueTuple(Of String, String)"
+  IL_0011:  callvirt   "Function Object.ToString() As String"
+  IL_0016:  call       "Sub System.Console.WriteLine(String)"
+  IL_001b:  ret
+}
+]]>)
+        End Sub
+
+        <Fact()>
+        Public Sub SimpleTupleTargetTyped002()
+
+            Dim verifier = CompileAndVerify(
+<compilation>
+    <file name="a.vb">
+Imports System
+Module C
+
+    Sub Main()
+        Dim x as (Func(Of integer), Func(of String)) = (Function() 42, Function() "hi")
```

Comment:
I will ad a test with the next change. It would be mostly testing the lambda conversion, but we want to see if that works when inside tuple.